src/compute/src/server.rs

```diff
@@ -79,15 +80,55 @@ pub async fn compute_node_serve(
     let stream_config = Arc::new(config.streaming.clone());
     let batch_config = Arc::new(config.batch.clone());
 
+    info!("current meta_addr is {}", opts.meta_address);
+
     // Register to the cluster. We're not ready to serve until activate is called.
-    let meta_client = MetaClient::register_new(
+
+    let mut client_res = MetaClient::register_new(
         &opts.meta_address,
         WorkerType::ComputeNode,
         &client_addr,
         config.streaming.worker_node_parallelism,
     )
-    .await
-    .unwrap();
+    .await;
+
+    // FIXME: even after initializing the meta_client we have to be careful when talking to meta
+    // every call to meta may fail if the current meta is not a leader
+    // Below code does not handle meta failover
+
+    tracing::info!(
```

Comment:
compute will connect against initial meta leader, but is unable to handle meta failover